lumispy/signals/cl_spectrum.py

```diff
@@ -33,49 +35,91 @@ class CLSpectrum(LumiSpectrum):
+
+        :param threshold: 'auto' or int
+            The derivative magnitude threshold above which to find spikes.
+            If `int` set the threshold value use for the detecting the spikes.
+            If `auto`, determine the threshold value as being the first zero
+            value in the histogram obtained from the
+            :py:meth:`~hyperspy.signals._signal1d.Signal1D.spikes_diagnosis`
+            method.
+        :param add_noise: bool
+            Whether to add noise to the interpolated part of the spectrum.
+            The noise properties defined in the Signal metadata are used if present,
+             otherwise 'poisson' shot noise is used as a default.
+        :param noise_type: str
+            By default 'poission' shoot noise is used if `add_noise` is True.
+            Noise types: "white", "heteroscedastic" or "poisson".
+        :param show_diagnosis_histogram: bool
+            Plot or not the derivative histogram to show the magnitude of the spikes present.
+        :param inplace: bool
             If False, a new signal object is created and returned. If True, the original signal object is modified.
+        :param luminescence_roi: array
+            The peak position and peak widths of the peaks in the luminescence spectrum.
+            In the form of an array of pairwise elements [[peak1_x, peak1_width], [peak2_x, peak2_width],...]
+            in the units of the signal axis. It creates a signal_mask protecting the peak regions.
+            To be used instead of `signal_mask`.
+        :param signal_mask: boolean array
+            Restricts the operation to the signal locations not marked as True (masked).
+        :param navigation_mask: boolean array
+            Restricts the operation to the navigation locations not marked as True (masked).
+        :param default_spike_width: int
+            Width over which to do the interpolation when removing all spike.
+        :param kwargs: dict
+            Keyword arguments pass to `hyperspy.signal.signal.BaseSignal.get_histogram`.
+
+        :return: None or CLSpectrum
+            Depends on inplace, returns or overwrites the CLSpectrum after spike removal.
         """
+        if luminescence_roi is not None and signal_mask is None:
+            signal_mask = self._make_signal_mask(luminescence_roi)
+
+        if show_diagnosis_histogram:
+            self.spikes_diagnosis(navigation_mask=navigation_mask, signal_mask=signal_mask,
+                                  **kwargs)
+        if inplace:
+            signal = self
+        else:
+            signal = self.deepcopy()
+
+        spikes_removal = SpikesRemoval(signal=signal, navigation_mask=navigation_mask, signal_mask=signal_mask,
+                                       threshold=threshold, default_spike_width=default_spike_width,
+                                       add_noise=add_noise,)
+
+        setattr(spikes_removal, 'noise_type', noise_type)
```

Comment:
Would it make sense to [set the noise in the metadata](https://hyperspy.readthedocs.io/en/latest/user_guide/signal.html#setting-the-noise-properties) instead?

And then rely on:
https://github.com/hyperspy/hyperspy/blob/1ff7468e41faa1094c189e4d7c182216efd41390/hyperspy/signal_tools.py#L1571-L1581

Reply:
Yes. Indeed